### PR TITLE
📝 Update docstring generator to include target branch for push

### PR DESCRIPTION
**Pull request by Drewniok**
## Description

The docstring generator workflow successfully runs when merging branches originating from ``upstream-main``. However, it fails to push docstrings during workflow execution for branches originating from ``fork-main`` due to detached branches.

This PR resolves the issue by specifying the target branch for pushing docstrings.

Thanks @simon1hofmann for helping to figure this out!

## Checklist:

- [ ] The pull request only contains commits that are related to it.
- [ ] I have added appropriate tests and documentation.
- [ ] I have added a changelog entry.
- [ ] I have created/adjusted the Python bindings for any new or updated functionality.
- [ ] I have made sure that all CI jobs on GitHub pass.
- [ ] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

**Comment by simon1hofmann** [comment]
Then the changes of #475 can probably be reverted right? 
`on: push` should be enough.

**Comment by simon1hofmann** [comment]
I think it's not possible to push to the branch like that, also `add-and-commit` should take care of that automatically.

**Comment by Drewniok** [comment]
ah great! Thank you!

**Comment by marcelwa** [comment]
@Drewniok @simon1hofmann does this work as intended now?

**Comment by Drewniok** [comment]
@marcelwa Not yet

**Comment by marcelwa** [comment]
Alright, keep me posted

File: .github/workflows/pyfiction-docstring-generator.yml [hunk 1]
```diff
@@ -20,6 +16,7 @@ jobs:
         uses: actions/checkout@v4
         with:
           submodules: recursive
+          fetch-depth: 0 # Fetch all history for all branches and tags
```
Comment (marcelwa) [inline .github/workflows/pyfiction-docstring-generator.yml:19]: Why is this important?

Comment (simon1hofmann) [reply .github/workflows/pyfiction-docstring-generator.yml:19]: Previously, the docstring generator only worked for braches forked directly from `fiction` and not, e.g., `Drewniok/fiction`, which lead to this error message:

Error: Error: fatal: You are not currently on a branch.
  To push the history leading to the current (detached HEAD)
  state now, use
  
      git push origin HEAD:<name-of-remote-branch>

Comment (marcelwa) [reply .github/workflows/pyfiction-docstring-generator.yml:19]: Understood. Many thanks!

**Pull request merged**